pkg/kubectl/cmd/proxy.go

```diff
@@ -83,14 +83,33 @@ func NewCmdProxy(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().StringP("address", "", "127.0.0.1", "The IP address on which to serve on.")
 	cmd.Flags().Bool("disable-filter", false, "If true, disable request filtering in the proxy. This is dangerous, and can leave you vulnerable to XSRF attacks, when used with an accessible port.")
 	cmd.Flags().StringP("unix-socket", "u", "", "Unix socket on which to run the proxy.")
+	cmd.Flags().StringP("curl", "", "", "Raw URI to request from the proxy server.")
```

Comment:
If I'm reading this correctly, you first start the proxy server with `kubectl proxy [options]` and then you do `kubectl proxy --curl=/foo` which will run against the running proxy, correct? I was under the impression from the original issue from @pwittrock we were going to do it all at once - if `--curl` were provided we start the proxy server, run the `curl` and tear down the proxy server. That said, I'm not against to have it like this, but in that case `--curl` is mutually exclusive with every other flag (not only `--unix-socket`) and we need to validate it.